### PR TITLE
[Meshery-UI] Add Support For Exporting Models as OCI Images

### LINKED ISSUE #11352: [UI] Settings --> Registry: Model Import and Model Export
Related to https://github.com/meshery/meshery/issues/9775

### Current Behavior
Meshery supports importing and exporting of individual models into an out of the registry in a few different formats, however, this functionality is only exposed in `mesheryctl model`, currently.

### Desired Behavior
The ability to import and export models using Meshery UI **- using the OCI format only.**

### Implementation
1. Add Import button in Settings --> Registry page to upload a model in OCI format.
1. Add Export button in Settings --> Registry page to export a model in OCI format.
---
Bonus:
3. Bulk export of more than one model at a time.

### Acceptance Tests
1. User can click "Import" to open a file browser and upload a model in OCI format to Meshery Server, which upon import, validates all of the models contained entities, registers valid entities, notifies the user with a toasted event  and one filed in the Notification Center that includes a summary of the number of entities successfully imported or details as to why the import failed and what to do about it (with MeshKit error reference)
2. User can select (checkmark) a single model name (that might contain one or more versions of the model). Click Export and have an OCI-formatted model file pushed to their client (web browser).

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by humblenginr**
**Notes for Reviewers**

This PR fixes #11352 
  

https://github.com/user-attachments/assets/db8e048f-0991-45af-9d3e-56c83a8cbf34

**DO NOT MERGE**
Needs to be refactored. And UI/UX has to be finalised. 
Depends on https://github.com/meshery/meshkit/pull/538

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by Jougan-0 — APPROVED** [review]

**Comment by Jougan-0** [comment]
Any update on import button. Are you working on it?

File: ui/components/MeshModelRegistry/MeshModelDetails.js [hunk 1]
```diff
@@ -157,10 +161,30 @@ const ModelContents = withSuppressedErrorBoundary(({ modelDef }) => {
...
     <div>
-      <div style={{ display: 'flex', justifyContent: 'space-between' }}>
+      <div style={{ display: 'flex', justifyContent: 'space-between', alignItems: 'center' }}>
         <TitleWithImg displayName={modelDef.displayName} iconSrc={modelDef?.metadata?.svgColor} />
+        {ExportAvailable ? (
+          <UsesSistent>
+            <Button
+              onClick={handleExport}
+              variant="contained"
+              startIcon={<DownloadIcon />}
```
Comment (leecalcote) [inline ui/components/MeshModelRegistry/MeshModelDetails.js:181]: Icon should be white (should support user-selected color theme).

@captain-Akshay can you assist here?

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: server/handlers/component_handler.go [hunk 2]
```diff
@@ -1178,6 +1179,59 @@ func prettifyCompDefSchema(entities []entity.Entity) []v1beta1.ComponentDefiniti
...
+// responses:
+//	200: []byte
+func (h *Handler) ExportModel(rw http.ResponseWriter, r *http.Request) {
+	modelId := r.URL.Query().Get("id")
+
+	// 1. Get the model data
+	modelFilter := &regv1beta1.ModelFilter{Id: modelId}
+	e, err := h.registryManager.GetEntityById(modelFilter)
+	if err != nil {
+		h.log.Error(ErrGetMeshModels(err)) //TODO: Add appropriate meshkit error
```
Comment (leecalcote) [inline server/handlers/component_handler.go:1199]: @humblenginr a pattern of short cutting is developing.

File: server/handlers/component_handler.go [hunk 3]
```diff
@@ -1178,6 +1179,59 @@ func prettifyCompDefSchema(entities []entity.Entity) []v1beta1.ComponentDefiniti
...
+	// 2. Convert it to oci
+	temp := os.TempDir()
+	modelDir := filepath.Join(temp, model.Name)
+	os.Mkdir(modelDir, 0700)
+	defer os.RemoveAll(modelDir)
+	err = model.WriteModelDefinition(filepath.Join(modelDir, "model.yaml"), "yaml")
+	// build oci image for the model
+	img, err := oci.BuildImage(modelDir)
+	if err != nil {
+		h.log.Error(ErrGetMeshModels(err)) //TODO: Add appropriate meshkit error
```
Comment (leecalcote) [inline server/handlers/component_handler.go:1213]: This makes for much back and forth on reviews. Dot your i's and cross your t's, please.

File: server/handlers/component_handler.go [hunk 4]
```diff
@@ -1178,6 +1179,59 @@ func prettifyCompDefSchema(entities []entity.Entity) []v1beta1.ComponentDefiniti
...
+		http.Error(rw, ErrGetMeshModels(err).Error(), http.StatusInternalServerError)
+	}
+	tarfileName := filepath.Join(modelDir, "model.tar")
+	err = oci.SaveOCIArtifact(img, tarfileName, model.Name)
```
Comment (leecalcote) [inline server/handlers/component_handler.go:1217]: Is archive compression happening here? If not, where is the `.gz` bit happening?

Comment (leecalcote) [reply server/handlers/component_handler.go:1217]: Ideally, we're being consistent with behavior used while exporting Designs (same function, same options). Using MeshKit to achieve this? https://github.com/meshery/meshkit/blob/fd9565f3bbbfb6731c25726714029913f82b3743/models/oci/utils.go#L27

**Comment by leecalcote** [comment]
Stopping by to review again. No new commits to review.

**Comment by leecalcote** [comment]
Stopping by to review again. No new commits to review.

**Comment by Jougan-0** [comment]
@humblenginr Update??? Could you please address the comments or shall we close and move forward in another PR.

**Comment by leecalcote** [comment]
No updates here. PR abandoned?

**Comment by humblenginr** [comment]
closing this in favour of https://github.com/meshery/meshery/pull/11354

**Pull request closed without merging**